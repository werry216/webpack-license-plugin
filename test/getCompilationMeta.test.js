const getCompilationMeta = require('../lib/getCompilationMeta')
const { sep } = require('path')

describe('getCompilationMeta', () => {
  test('returns sorted, flattened and unique package meta for all chunks', () => {
    const base = ['home', 'codepunkt', 'projects', 'monorepo'].join(sep)
    const compilation = {
      chunks: [
        {
          modules: [
            [base, 'packages', 'client', 'src'].join(sep),
            [base, 'node_modules', 'package-b'].join(sep),
            [base, 'node_modules', '@scope', 'package-d'].join(sep),
            [base, 'node_modules', '@scope', 'package-c', 'dist'].join(sep),
          ].map(context => ({ context })),
        },
        {
          modules: [
            [base, 'node_modules', 'package-a', 'dist'].join(sep),
            [base, 'packages', 'client'].join(sep),
            [base, 'node_modules', '@scope', 'package-d'].join(sep),
            [
              base,
              'node_modules',
              'package-e',
              'node_modules',
              'package-f',
            ].join(sep),
          ].map(context => ({ context })),
        },
      ],
    }

    expect(getCompilationMeta(compilation)).toEqual([
      {
        name: '@scope/package-c',
        path: [base, 'node_modules', '@scope', 'package-c'].join(sep),
      },
      {
        name: '@scope/package-d',
        path: [base, 'node_modules', '@scope', 'package-d'].join(sep),
      },
      {
        name: 'package-a',
        path: [base, 'node_modules', 'package-a'].join(sep),
      },
      {
        name: 'package-b',
        path: [base, 'node_modules', 'package-b'].join(sep),
      },
      {
        name: 'package-f',
        path: [
          base,
          'node_modules',
          'package-e',
          'node_modules',
          'package-f',
        ].join(sep),
      },
    ])
  })
})
