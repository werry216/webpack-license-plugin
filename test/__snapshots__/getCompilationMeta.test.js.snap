// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCompilationMeta returns sorted, flattened and unique package meta for all chunks 1`] = `
Array [
  Object {
    "name": "@scope/package-c",
    "path": "home/codepunkt/projects/monorepo/node_modules/@scope/package-c",
  },
  Object {
    "name": "@scope/package-d",
    "path": "home/codepunkt/projects/monorepo/node_modules/@scope/package-d",
  },
  Object {
    "name": "package-a",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-a",
  },
  Object {
    "name": "package-b",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-b",
  },
  Object {
    "name": "package-f",
    "path": "home/codepunkt/projects/monorepo/node_modules/package-e/node_modules/package-f",
  },
]
`;
